fetchmap/index.ts

```diff
@@ -0,0 +1,265 @@
+        const currentDeckLayers = (deck.props.layers || []) as Layer[];
+        const newLayers = currentDeckLayers.map((layer: Layer) => {
+          if (layer && layer.id === layerId) {
+            return layer.clone({visible});
+          }
+          return layer;
+        });
+        deck.setProps({layers: newLayers});
+      });
+    }
+
+    // Finally, update the deck.gl view state and layers
+    deck.setProps({
+      initialViewState: finalInitialViewState,
+      layers: LayerFactory(currentMapData.layers)
+    });
+
+    map.jumpTo({
+      center: [finalInitialViewState.longitude, finalInitialViewState.latitude],
+      zoom: finalInitialViewState.zoom,
+      bearing: finalInitialViewState.bearing,
+      pitch: finalInitialViewState.pitch
+    });
+  } catch (error) {
+    if (mapNameEl) mapNameEl.innerHTML = 'Error loading map(s)';
+    if (layerCountEl) layerCountEl.innerHTML = 'Error';
+    currentMapData = null; // Clear data on error
+    deck.setProps({layers: []}); // Clear layers on error
+  }
+}
+
+// Initialize map selector and load default map(s)
+function setupMapSelector() {
+  if (!mapSelectorContainer || !loadMapsButton) {
+    // Fallback: Load the first map by default if UI elements are missing but configs exist
+    if (MAP_CONFIGS.length > 0) {
+      initialize([MAP_CONFIGS[0].id]);
+    } else {
+      initialize([]); // Show "No Map Selected" or similar
+    }
+    return;
+  }
+
+  MAP_CONFIGS.forEach((mapConfig, index) => {
+    const checkbox = document.createElement('input');
+    checkbox.type = 'checkbox';
+    // Ensure unique ID for each checkbox, useful for labels
+    checkbox.id = `map-checkbox-${mapConfig.id}-${index}`;
+    checkbox.value = mapConfig.id;
+    checkbox.name = 'mapSelection';
+
+    const label = document.createElement('label');
+    label.htmlFor = checkbox.id;
+    label.textContent = mapConfig.name;
+
+    const div = document.createElement('div');
+    div.className = 'map-selector-option';
+    div.appendChild(checkbox);
+    div.appendChild(label);
+    mapSelectorContainer.appendChild(div);
+  });
+
+  mapSelectorContainer.appendChild(loadMapsButton);
+
+  loadMapsButton.addEventListener('click', () => {
+    const selectedCheckboxes = document.querySelectorAll<HTMLInputElement>(
+      'input[name="mapSelection"]:checked'
+    );
+    const selectedIds = Array.from(selectedCheckboxes).map(cb => cb.value);
+    initialize(selectedIds);
+  });
+
+  // Load the first map by default on initial page load
+  const firstCheckbox =
+    mapSelectorContainer.querySelector<HTMLInputElement>('input[type="checkbox"]');
+  if (firstCheckbox) {
+    firstCheckbox.checked = true;
+    initialize([firstCheckbox.value]);
+  } else if (MAP_CONFIGS.length > 0) {
+    // Fallback if somehow checkboxes weren't created but configs exist
+    initialize([MAP_CONFIGS[0].id]);
+  } else {
+    initialize([]); // No maps to load if no configs
```

Comment:
I might vote to remove the "if no configs" checks to simplify things, it feels like it distracts from the example a bit.